packages/controllers/src/plugins/PluginController.ts

```diff
@@ -53,38 +53,56 @@ type RemoveAllPermissionsFunction = (pluginIds: string[]) => void;
 type CloseAllConnectionsFunction = (domain: string) => void;
 type RequestPermissionsFunction = (
   domain: string,
-  requestedPermissions: IRequestedPermissions,
-) => IOcapLdCapability[];
+  requestedPermissions: RequestedPluginPermissions,
+) => Promise<IOcapLdCapability[]>;
 type HasPermissionFunction = (
   domain: string,
   permissionName: string,
 ) => boolean;
 type GetPermissionsFunction = (domain: string) => IOcapLdCapability[];
+type TerminateWorkerOf = (pluginName: string) => void;
+type Command = (
+  workerId: string,
+  message: JsonRpcRequest<unknown>,
+) => Promise<unknown>;
+type TerminateAll = () => void;
+type CreatePluginWorker = (metadate: PluginWorkerMetadata) => Promise<string>;
+type StartPlugin = (
+  workerId: string,
+  pluginData: PluginData,
+) => Promise<unknown>;
 
-interface StoredPlugins {
-  [pluginId: string]: Plugin;
-}
+type StoredPlugins = Record<string, Plugin>;
 
-export interface PluginControllerState {
+export type PluginControllerState = {
+  inlinePluginIsRunning: boolean;
   plugins: StoredPlugins;
-  pluginStates: Record<string, unknown>;
-}
+  pluginStates: {
+    [id: string]: Json;
+  };
+};
 
 export interface PluginControllerMemState {
   inlinePluginIsRunning: boolean;
-  plugins: { [pluginId: string]: SerializablePlugin };
-  pluginStates: Record<string, unknown>;
+  plugins: Record<string, SerializablePlugin>;
+  pluginStates: {
+    [id: string]: Json;
+  };
 }
 
 interface PluginControllerArgs {
-  initState: Partial<PluginControllerState>;
+  messenger: RestrictedControllerMessenger<string, any, any, string, string>;
+  state: PluginControllerState;
   removeAllPermissionsFor: RemoveAllPermissionsFunction;
-  setupWorkerPluginProvider: SetupWorkerConnection;
   closeAllConnections: CloseAllConnectionsFunction;
   requestPermissions: RequestPermissionsFunction;
   getPermissions: GetPermissionsFunction;
   hasPermission: HasPermissionFunction;
-  workerUrl: URL;
+  terminateWorkerOf: TerminateWorkerOf;
+  command: Command;
+  terminateAll: TerminateAll;
+  createPluginWorker: CreatePluginWorker;
+  startPlugin: StartPlugin;
```

Comment:
Is the idea that we'll eventually extract these into that generic execution environment service we discussed?